PROTO_tests/tests/client_dtypes_test.py

```diff
@@ -0,0 +1,181 @@
+            f = ak.Fields(values, names)
+
+        names = ["t", "t"]
+        with pytest.raises(ValueError):
+            f = ak.Fields(values, names)
+
+        names = ["t", ""]
+        with pytest.raises(ValueError):
+            f = ak.Fields(values, names)
+
+        names = ["abc", "123"]
+        with pytest.raises(ValueError):
+            f = ak.Fields(values, names, separator="abc")
+
+        with pytest.raises(ValueError):
+            f = ak.Fields(values, names)
+
+        with pytest.raises(ValueError):
+            f = ak.Fields(values, names, pad="|!~", separator="//")
+
+    @pytest.mark.parametrize("prob_size", pytest.prob_size)
+    @pytest.mark.parametrize("int_types", INT_TYPES)
+    def test_ipv4_creation(self, prob_size, int_types):
+        ips = ak.randint(1, 2**32, prob_size)
+        ip_list = ak.array(ips, dtype=int_types)
+        ipv4 = ak.IPv4(ip_list)
+        py_ips = [ipaddress.IPv4Address(ip).compressed for ip in ips.to_list()]
+
+        assert isinstance(ipv4, ak.IPv4)
+        assert ipv4.to_list() == py_ips
+        assert ipv4.dtype == ak.bitType
+
+        with pytest.raises(TypeError):
+            ipv4 = ak.IPv4(f"{ips[0]}")
+        with pytest.raises(TypeError):
+            ipv4 = ak.IPv4(ak.array([ips[0] + 0.177]))
+
+        # Test handling of python dotted-quad input
+        ipv4 = ak.ip_address(py_ips)
+        assert isinstance(ipv4, ak.IPv4)
+        assert ipv4.to_list() == py_ips
+        assert ipv4.dtype == ak.bitType
+
+    @pytest.mark.parametrize("prob_size", pytest.prob_size)
+    def test_ipv4_normalization(self, prob_size):
+        prob_size = 100
+        ips = ak.randint(1, 2**32, prob_size)
+        ip_list = ak.array(ips)
+        ipv4 = ak.IPv4(ip_list)
+        ip_as_dot = [ipaddress.IPv4Address(ip) for ip in ips.to_list()]
+        ip_as_int = [ipv4.normalize(ipd) for ipd in ip_as_dot]
+        assert ips.to_list() == ip_as_int
+
+    @pytest.mark.parametrize("size", pytest.prob_size)
+    def test_is_ipv4(self, size):
+        x = [random.getrandbits(32) for i in range(size)]
+
+        ans = ak.is_ipv4(ak.array(x, dtype=ak.uint64))
+        assert ans.to_list() == [True] * size
+
+        ipv4 = ak.IPv4(ak.array(x))
+        assert ak.is_ipv4(ipv4).to_list() == [True] * size
+
+        x = [random.getrandbits(64) if i < size / 2 else random.getrandbits(32) for i in range(size)]
+        ans = ak.is_ipv4(ak.array(x, ak.uint64))
+        assert ans.to_list() == [i >= size / 2 for i in range(size)]
+
+        with pytest.raises(TypeError):
+            ak.is_ipv4(ak.array(x, ak.float64))
+
+        with pytest.raises(RuntimeError):
+            ak.is_ipv4(ak.array(x, dtype=ak.uint64), ak.arange(2, dtype=ak.uint64))
+
+    @pytest.mark.parametrize("size", pytest.prob_size)
+    def test_is_ipv6(self, size):
+        x = [random.getrandbits(128) for _ in range(size)]
+        low = ak.array([i & (2**64 - 1) for i in x], dtype=ak.uint64)
```

Comment:
same deal as with `test_is_ipv4`, with all the converting between client and server i vote we pick a fixed size for these arrays

Reply:
Updated